### PR TITLE
All /R commands cache gadgets and use rop.db

### LINKED ISSUE #5393: Honor rop.db
right now requires asm.esil=true and only works with  /Rl

### PR DESCRIPTION
**Pull request by Darredevil**
Fixed #5393 and #5394

### CODE REVIEW

File: libr/core/cmd_search.c [hunk 1]
```diff
@@ -916,11 +916,13 @@ static void print_rop (RCore *core, RList *hitlist, char mode, bool *json_first)
 	unsigned int size = 0;
 	RAnalOp analop = {0};
 	RAsmOp asmop;
+	Sdb *db = sdb_ns (core->sdb, "rop", false);
```
Comment (radare) [inline libr/core/cmd_search.c]: why false?

Comment (radare) [inline libr/core/cmd_search.c]: check if db is null after that call, if so just show an error

File: libr/core/cmd_search.c [hunk 2]
```diff
@@ -1020,6 +1027,12 @@ static void print_rop (RCore *core, RList *hitlist, char mode, bool *json_first)
 			}
 			free (buf);
 		}
+		if (rop_db && hit) {
```
Comment (radare) [inline libr/core/cmd_search.c]: if (db && ...

File: libr/core/cmd_search.c (hunk 1 above)
Comment (radare) [inline libr/core/cmd_search.c]: only create the db if its suposed to be used. so move the conditionals up to here

**Comment by Darredevil** [comment]
@radare fixed comments, should be ok now

**Comment by radare** [comment]
looks good now. thanks!

**Pull request merged**